### PR TITLE
Use iam to get tls config

### PR DESCRIPTION
**Pull request by MykolaSolyanko**

### CODE REVIEW

File: iamclient/iamclient_test.go [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+	"github.com/aoscloud/aos_updatemanager/iamclient"
+
+	"github.com/aoscloud/aos_updatemanager/config"
+)
+
+/***********************************************************************************************************************
+ * Consts
+ **********************************************************************************************************************/
+
+const serverURL = "localhost:8089"
```
Comment (Eabramov84) [inline iamclient/iamclient_test.go:41]: This is just  a test, but iam public has 8090 port
https://kb.epam.com/display/EPMDAEPRA/Core

**Review by Eabramov84 — CHANGES_REQUESTED** [review]

File: aos_updatemanager.cfg [hunk 2]
```diff
@@ -2,9 +2,10 @@
     "ID": "um",
     "ServerUrl": "localhost:9000",
     "CACert": "/etc/ssl/certs/rootCA.crt",
-    "CertStorage": "/var/aos/crypt/um/",
+    "CertStorage": "um",
```
Comment (Eabramov84) [inline aos_updatemanager.cfg:5]: Move changes retated CertStorage to separate commit

File: iamclient/iamclient.go [hunk 3]
```diff
@@ -0,0 +1,143 @@
+// SPDX-License-Identifier: Apache-2.0
```
Comment (Eabramov84) [inline iamclient/iamclient.go:1]: Maybe there is no sense to keep a connection to IAM. We need this connection only on startup to get key, cert URL.

File: aos_updatemanager.cfg [hunk 4]
```diff
@@ -2,9 +2,10 @@
...
     "DownloadDir": "/var/aos/updatemanager/download",
     "WorkingDir": "/var/aos/updatemanager",
+    "iamServerUrl": "aosiam:8089",
```
Comment (Eabramov84) [inline aos_updatemanager.cfg:8]: **I**amServerUrl

File: iamclient/iamclient.go [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+//
+// Copyright (C) 2021 Renesas Electronics Corporation.
```
[the remark this answers is not in the document]
Comment (MykolaSolyanko) [reply iamclient/iamclient.go]: Done

File: iamclient/iamclient.go [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+ * Types
+ **********************************************************************************************************************/
+
+// Client IAM client instance.
+type Client struct {
+	sync.Mutex
+
+	connection    *grpc.ClientConn
+	pbPublic      pb.IAMPublicServiceClient
+	cryptoContext *cryptutils.CryptoContext
```
[the remark this answers is not in the document]
Comment (MykolaSolyanko) [reply iamclient/iamclient.go]: Done

File: iamclient/iamclient.go [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+// GetCertificate gets certificate by issuer.
+func (client *Client) GetCertificate(cerType string) (certURL, keyURL string, err error) {
+	log.WithFields(log.Fields{
+		"type": cerType,
+	}).Debug("Get certificate")
+
+	ctx, cancel := context.WithTimeout(context.Background(), iamRequestTimeout)
+	defer cancel()
+
+	request := &pb.GetCertRequest{Type: cerType}
```
[the remark this answers is not in the document]
Comment (MykolaSolyanko) [reply iamclient/iamclient.go]: Done

File: iamclient/iamclient_test.go [hunk 8]
```diff
@@ -0,0 +1,184 @@
+// SPDX-License-Identifier: Apache-2.0
+//
+// Copyright (C) 2021 Renesas Electronics Corporation.
```
[the remark this answers is not in the document]
Comment (MykolaSolyanko) [reply iamclient/iamclient_test.go]: Done

File: iamclient/iamclient_test.go (hunk 1 above)
Comment (MykolaSolyanko) [reply iamclient/iamclient_test.go]: Done

File: aos_updatemanager.cfg (hunk 2 above)
Comment (MykolaSolyanko) [reply aos_updatemanager.cfg:5]: Done

File: iamclient/iamclient.go (hunk 3 above)
Comment (MykolaSolyanko) [reply iamclient/iamclient.go:1]: Done

File: aos_updatemanager.cfg (hunk 4 above)
Comment (MykolaSolyanko) [reply aos_updatemanager.cfg]: Done

File: iamclient/iamclient.go [hunk 9]
```diff
@@ -0,0 +1,132 @@
...
+// GetCertificate gets certificate by issuer.
+func (client *Client) GetCertificate(
+	cerType string, cryptocontext *cryptutils.CryptoContext) (certURL, keyURL string, err error) {
+	log.WithFields(log.Fields{
+		"type": cerType,
+	}).Debug("Get certificate")
+
+	connection, pbPublic, err := client.createConnection(cryptocontext)
+	if err != nil {
+		return "", "", aoserrors.Wrap(err)
```
Comment (Eabramov84) [inline iamclient/iamclient.go:70]: this wrap could be omitted, because createConnection method already returned wrapped error

File: iamclient/iamclient.go [hunk 10]
```diff
@@ -0,0 +1,132 @@
...
+	ctx, cancel := context.WithTimeout(context.Background(), iamRequestTimeout)
+	defer cancel()
+
+	var secureOpt grpc.DialOption
+
+	if client.insecure {
+		secureOpt = grpc.WithInsecure()
+	} else {
+		if cryptocontext == nil {
+			return nil, nil, aoserrors.New("CryptoContext must not be nil")
```
Comment (Eabramov84) [inline iamclient/iamclient.go:111]: error should start from small letter

File: iamclient/iamclient.go [hunk 11]
```diff
@@ -0,0 +1,132 @@
...
+
+		tlsConfig, err := cryptocontext.GetClientTLSConfig()
+		if err != nil {
+			return nil, nil, aoserrors.Wrap(err)
+		}
+
+		secureOpt = grpc.WithTransportCredentials(credentials.NewTLS(tlsConfig))
+	}
+
+	connection, err = grpc.DialContext(ctx, client.iamServerURL, secureOpt, grpc.WithBlock())
```
Comment (Eabramov84) [inline iamclient/iamclient.go:122]: combine with onelin
if connection, err = ...; err!=nil

File: iamclient/iamclient_test.go [hunk 12]
```diff
@@ -0,0 +1,183 @@
...
+}
+
+/***********************************************************************************************************************
+ * Main
+ **********************************************************************************************************************/
+
+func TestMain(m *testing.M) {
+	var err error
+
+	tmpDir, err = ioutil.TempDir("", "iam_")
```
Comment (Eabramov84) [inline iamclient/iamclient_test.go:88]: combine in one line

File: iamclient/iamclient.go (hunk 9 above)
Comment (MykolaSolyanko) [reply iamclient/iamclient.go]: Done

File: iamclient/iamclient.go (hunk 10 above)
Comment (MykolaSolyanko) [reply iamclient/iamclient.go]: Done

File: iamclient/iamclient.go (hunk 11 above)
Comment (MykolaSolyanko) [reply iamclient/iamclient.go]: Done

File: iamclient/iamclient_test.go (hunk 12 above)
Comment (MykolaSolyanko) [reply iamclient/iamclient_test.go]: Done

File: .golangci.yml [hunk 13]
```diff
@@ -10,6 +10,7 @@ linters:
     - cyclop
     - nestif
     - ireturn
+    - gomoddirectives
```
Comment (Eabramov84) [inline .golangci.yml:13]: change commit message : 
.. Disable gomoddirectives lint

File: umclient/umclient.go [hunk 14]
```diff
@@ -195,15 +203,21 @@ func (status ComponentStatus) String() string {
...
 	var secureOpt grpc.DialOption
 
 	if insecure {
 		secureOpt = grpc.WithInsecure()
 	} else {
-		tlsConfig, err := cryptutils.GetClientMutualTLSConfig(config.CACert, config.CertStorage)
+		certURL, keyURL, err := provider.GetCertificate(config.CertStorage, cryptocontext)
+		if err != nil {
+			return aoserrors.Wrap(err)
+		}
```
[the remark this answers is not in the document]
Comment (MykolaSolyanko) [reply umclient/umclient.go:219]: Done

File: .golangci.yml (hunk 13 above)
Comment (MykolaSolyanko) [reply .golangci.yml:13]: Done

**Review by Eabramov84 — APPROVED** [review]
